garage/envs/util.py

```diff
@@ -31,7 +34,10 @@ def flatten(space, obs):
         return np.concatenate(
             [flatten(c, xi) for c, xi in zip(space.spaces, obs)])
     else:
-        raise NotImplementedError
+        try:
```

Comment:
This results in magical behavior which will hide bugs.

If a not ImplementedError was encountered here, then the bug lives somewhere else in the codebase. It seems likely someone has attempted to flatten a `garage.Space` where they should only posess a gym.space